<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [12]:
# Membuat dataframes dari dictionary
import pandas as pd

data = {
    'Nama': ['Rofiq', 'Krisna', 'Aiza', 'Rifqi'],
    'Umur': [20, 20, 5, 31],
    'Profesi': ['Data Scientist', 'UI/UX Designer', '-', 'Software Engineer'],
    'Hobi': ['Design', 'Ngoding', 'Memasak', 'Nigh Ride']
}

df = pd.DataFrame(data)

print(df)


     Nama  Umur            Profesi       Hobi
0   Rofiq    20     Data Scientist     Design
1  Krisna    20     UI/UX Designer    Ngoding
2    Aiza     5                  -    Memasak
3   Rifqi    31  Software Engineer  Nigh Ride


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [13]:
# Membaca data dan memasukannya kedalam suatu variabel
data = pd.read_csv('C:\Bootcamp-Python DS\Project Python DS\src\Weekly Project\Week 2\diamonds.csv')

print(data)


       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
- Apa perbedaan Series dengan list dan dictionary
  - Series:
    - Definisi: Series adalah struktur data satu dimensi dari pustaka pandas yang mirip dengan array atau list, tetapi dengan label yang dapat diatur (index).
    - Indeks: Setiap elemen dalam Series memiliki indeks yang memudahkan akses dan manipulasi data.
    - Fungsi: Memiliki banyak metode dan fungsi bawaan untuk analisis data seperti agregasi dan statistik.

  - List:
    - Definisi: list adalah struktur data bawaan di Python yang merupakan urutan elemen yang dapat diubah (mutable) dan dapat berisi berbagai tipe data.
    - Indeks: Menggunakan indeks numerik yang dimulai dari 0.
    - Fungsi: Metode terbatas dibandingkan dengan Series, seperti append(), remove(), dan sort().

  - Dictionary:
    - Definisi: dictionary adalah struktur data bawaan di Python yang menyimpan pasangan kunci-nilai (key-value).
    - Indeks: Menggunakan kunci (key) yang unik untuk mengakses nilai (value).
    - Fungsi: Metode seperti get(), items(), dan keys() untuk manipulasi data.

- Apa perbedaan dataframe dan series
  - Series:
    - Definisi: Series adalah struktur data satu dimensi yang menyimpan data dengan label indeks.
    - Dimensi: Hanya satu dimensi (seperti kolom dalam tabel).
    - Penggunaan: Biasanya digunakan untuk menyimpan satu kolom data.

  - DataFrame:
    - Definisi: DataFrame adalah struktur data dua dimensi dari pustaka pandas yang mirip dengan tabel database atau lembar kerja Excel.
    - Dimensi: Dua dimensi (baris dan kolom).
    - Penggunaan: Digunakan untuk menyimpan dan menganalisis data tabular yang lebih kompleks dengan banyak kolom dan baris.

In [14]:
# code here
nilai_mata_pelajaran = pd.Series([85, 90, 78, 92], index=['Matematika', 'Fisika', 'Kimia', 'Biologi'])
print("Nilai Mata Pelajaran:")
print(nilai_mata_pelajaran)

cuaca_harian = pd.Series({
    'Senin': 'Cerah',
    'Selasa': 'Hujan',
    'Rabu': 'Cerah',
    'Kamis': 'Berawan',
    'Jumat': 'Cerah'
})
print("\nCuaca Harian:")
print(cuaca_harian)

rak_buku = pd.Series([15, 22, 30, 8], index=['Rak A', 'Rak B', 'Rak C', 'Rak D'])
print("\nJumlah Buku di Setiap Rak:")
print(rak_buku)

Nilai Mata Pelajaran:
Matematika    85
Fisika        90
Kimia         78
Biologi       92
dtype: int64

Cuaca Harian:
Senin       Cerah
Selasa      Hujan
Rabu        Cerah
Kamis     Berawan
Jumat       Cerah
dtype: object

Jumlah Buku di Setiap Rak:
Rak A    15
Rak B    22
Rak C    30
Rak D     8
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [15]:
# code here
# Menampilkan 5 data pertama
print("5 data pertama:")
print(data.head())

# Menampilkan 5 data terakhir
print("5 data terakhir:")
print(data.tail())

# Menangambil nilai baris pertama dari kolom price
first_price = data['price'].iloc[0]
print("\nBaris pertama dari kolom 'price':", first_price)

# Mengambil nilai baris terakhir dari kolom color
last_color = data['color'].iloc[-1]
print("Baris terakhir dari kolom 'color':", last_color)


5 data pertama:
   Unnamed: 0  carat      cut color clarity  depth  table  price     x     y  \
0           1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98   
1           2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84   
2           3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07   
3           4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23   
4           5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35   

      z  
0  2.43  
1  2.31  
2  2.31  
3  2.63  
4  2.75  
5 data terakhir:
       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75  

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 
Statistik Deskriptif adalah cabang dari statistik yang berfokus pada pengumpulan, penyajian, dan pengorganisasian data sehingga dapat disajikan dengan cara yang mudah dipahami. Tujuan utamanya adalah untuk merangkum dan menjelaskan fitur utama dari suatu dataset dengan cara yang ringkas dan informatif.

- Contoh Penerapan: Misalkan kita memiliki dataset yang berisi nilai ujian siswa. Statistik deskriptif dapat digunakan untuk:
    - Menentukan rata-rata nilai ujian siswa (mean).
    - Mengukur seberapa bervariasinya nilai ujian siswa (standar deviasi).
    - Mengidentifikasi nilai tengah dari dataset (median).
    - Melihat apakah ada nilai yang sering muncul (modus).

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [16]:
# code here
# Menampilkan deskriptif statistik dari data
print(data.describe())

# Nilai rata-rata dari column price
mean_price = data['price'].mean()
print(f'Nilai rata-rata dari kolom price: {mean_price}')

# Nilai standar deviasi dari column depth
std_depth = data['depth'].std()
print(f'Nilai standar deviasi dari kolom depth: {std_depth}')

# Nilai maksimum dari column carat
max_carat = data['carat'].max()
print(f'Nilai maksimum dari kolom carat: {max_carat}')

         Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000     

Jawab:
- Nilai rata-rata dari kolom price: 3932.799721913237
- Nilai standar deviasi dari kolom depth: 1.432621318833661
- Nilai maksimum dari kolom carat: 5.01